chain/store/store.go

```diff
@@ -52,6 +58,81 @@ func NewLedgerStore() (*ChainStore, error) {
 	return chain, nil
 }
 
+func (cs *ChainStore) PrepareFastSync(fastSyncHeight uint32) error {
+	err := cs.persistSyncRootHeight(fastSyncHeight)
+	if err != nil {
+		return err
+	}
+
+	fastSyncStatusBuf := make([]byte, 4)
+	binary.LittleEndian.PutUint32(fastSyncStatusBuf[:], beginFastSync)
+	err = cs.st.BatchPut(db.TrieFastSyncStatusKey(), fastSyncStatusBuf)
+	if err != nil {
+		return err
+	}
+	err = cs.st.BatchCommit()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (cs *ChainStore) FastSyncDone(syncRootHash common.Uint256, fastSyncHeight uint32) error {
+	sdb, err := NewStateDB(syncRootHash, cs)
+	if err != nil {
+		return err
+	}
+
+	err = sdb.trie.TryTraverse(false)
+	if err != nil {
+		return err
+	}
+
+	hs := sdb.trie.Hash()
+	if syncRootHash.CompareTo(hs) != 0 {
+		log.Fatal("State root not equal, should be: %v, we got: %v", syncRootHash.ToHexString(), hs.ToHexString())
+	}
+	cs.States = sdb
+
+	refCountHeightBuffer := make([]byte, 4)
+	binary.LittleEndian.PutUint32(refCountHeightBuffer[:], fastSyncHeight-1)
+	err = cs.st.BatchPut(db.TrieRefCountHeightKey(), refCountHeightBuffer)
+	if err != nil {
+		return err
+	}
+
+	pruneHeightBuffer := make([]byte, 4)
```

Comment:
Better to create a refcount and use the existing `PersistPrunedHeights()` method